src/v/kafka/requests/fetch_request.cc

```diff
@@ -332,6 +332,16 @@ static ss::future<read_result> read_from_partition(
       });
 }
 
+std::optional<partition_wrapper> make_partition_wrapper(
+  const model::materialized_ntp& mntp,
+  ss::lw_shared_ptr<cluster::partition> partition,
+  std::optional<storage::log> log) {
+    if (mntp.is_materialized() && !log) {
+        return std::nullopt;
+    }
+    return partition_wrapper(partition, log);
```

Comment:
if `mntp` is not materialized, then `log` contains the result of `mgr.log(mntpv.input_ntp()` which is the log of ntp that the client is targeting. since that log will exist, will this create a partition wrapper that reads directly from the log instead of going through raft?

Reply:
Thats totally correct, nice catch. Interesting how no unit tests failed in this scenario, how could that be??

Reply:
because going through raft changes the visibility range of the log offsets. the tests aren't that detailed.